webapp/channels/src/components/user_settings/modal/user_settings_modal.tsx

```diff
@@ -84,6 +98,22 @@ class UserSettingsModal extends React.PureComponent<Props, State> {
 
     componentDidMount() {
         document.addEventListener('keydown', this.handleKeyDown);
+
+        if (this.props.adminMode && this.props.userID) {
+            this.setState({loading: true});
+
+            if (!this.props.userPreferences) {
+                this.props.actions.getUserPreferences(this.props.userID);
```

Comment:
Would it be possible to only load the settings that this component cares about? We know that we have some problems with loading a lot of preferences on initial load, so perhaps we could only load certain categories of preferences. The ability to do that is actually why preferences categories exist, but that's barely used anywhere 😛

Reply:
The API doesn't support specifying the categories and there are dozens of categories of preferences the whole modal needs. This is beyond the scope for now. Skipping.

Reply:
Ah, I thought we might've gotten lucky there with all the settings being in the `display` or `advanced` category. I guess if we find this to be an issue down the line, we'll have to add a better API for it